main.tf

```
@@ -359,7 +359,7 @@ resource "aws_subnet" "database" {
 resource "aws_db_subnet_group" "database" {
   count = var.create_vpc && length(var.database_subnets) > 0 && var.create_database_subnet_group ? 1 : 0
 
-  name        = lower(var.name)
+  name        = var.database_subnet_group_name != null ? lower(var.database_subnet_group_name) : lower(var.name)
```

I think we should be able to get away with:
```suggestion
  name        = coalesce(lower(var.database_subnet_group_name, var.name))
```